language/move-prover/bytecode-to-boogie/src/spec_translator.rs

```diff
@@ -860,14 +862,45 @@ impl<'env> SpecTranslator<'env> {
                     res = format!("Dereference(__m, {})", res);
                     t = *vt;
                 }
+
                 for f_or_i in fields_and_indices {
-                    if let FieldOrIndex::Field(f) = f_or_i {
-                        let (s, tf) = self.translate_field_access(&t, f);
-                        res = format!("SelectField({}, {})", res, s);
-                        t = tf;
-                    } else {
-                        // TODO: Implement vector index.  For now, generates illegal Boogie.
-                        res = format!("Unimplemented vector index. AST = {:?}", f_or_i);
+                    match f_or_i {
+                        FieldOrIndex::Field(f) => {
+                            let (s, tf) = self.translate_field_access(&t, f);
+                            res = format!("SelectField({}, {})", res, s);
+                            t = tf;
+                        }
+                        FieldOrIndex::Index(i) => {
+                            match i {
+                                SpecExp::Update(ind, val) => {
+                                    let BoogieExpr(idx, _t_ind) = self.translate_expr( & ind);
+                                    let BoogieExpr(val, _t_val) = self.translate_expr( & val);
+                                    // TODO: raise an error if t_ind is not an integer type (u8, u64, u128)
+                                    // TODO: raise an error if t_val is not an Element type
+                                    res = format !("update_vector({}, i#Integer({}), {})", res, idx, val);
```

Comment:
No space after `format` and before `!` to follow rust style.

Reply:
BTW, have you autoformatted the code? (can be done with CLion, or by running `cargo fmt`). All code should be autoformatted. This may remove glitches like spaces which aren't following Rust style guidelines. (Above I see also `( & val)` which isn't rust style.)

Reply:
I fixed them with `cargo fmt`. Thanks for the info.